crates/block-producer/src/block_builder/mod.rs

```diff
@@ -112,8 +121,10 @@ impl BlockBuilder {
         self.get_block_inputs(selected)
             .inspect_ok(BlockBatchesAndInputs::inject_telemetry)
             .and_then(|inputs| self.propose_block(inputs))
-            .inspect_ok(ProposedBlock::inject_telemetry)
-            .and_then(|inputs| self.prove_block(inputs))
+            .inspect_ok(|(proposed_block, _)| {
+                ProposedBlock::inject_telemetry(proposed_block);
+            })
+            .and_then(|(proposed_block, inputs)| self.validate_and_prove_block(proposed_block, inputs))
```

Comment:
@bobbinth Is this still where the proving flow will run? Do we want to remove proving from here altogether? And move it to the store?

Reply:
We do, but I would do that in a separate PR because this would entail quite a few more changes.